RileyLinkKit/PumpOpsSynchronous.swift

```diff
@@ -335,6 +335,12 @@ class PumpOpsSynchronous {
 
         var events = [TimestampedHistoryEvent]()
         var timeAdjustmentInterval: NSTimeInterval = 0
+        
+        // Going to scan backwards in time through events, so event time should be monotonically decreasing
+        // Start with some time in the future, to account for the condition when the pump's clock is ahead
+        // of ours by a small amount.
+        var timeCursor = NSDate(timeIntervalSinceNow: NSTimeInterval(minutes: 60))
+        let eventTimestampDeltaAllowance = NSTimeInterval(minutes: 60)
```

Comment:
Are you sure it doesn't show up during delivery? I believe it does on the 523. 

Re: reservoir, LoopKit always prefers reservoir over history unless the reservoir data is deemed incomplete. Even so, since the square wave prevents temp basal adjustments, it wouldn't be able to make changes based on incorrect data. 

> On Aug 29, 2016, at 12:09 PM, Pete Schwamb <notifications@github.com> wrote:
> 
> In RileyLinkKit/PumpOpsSynchronous.swift:
> 
> > @@ -335,6 +335,12 @@ class PumpOpsSynchronous {
> >  
> >          var events = [TimestampedHistoryEvent]()
> >          var timeAdjustmentInterval: NSTimeInterval = 0
> > +        
> > +        // Going to scan backwards in time through events, so event time should be monotonically decreasing
> > +        // Start with some time in the future, to account for the condition when the pump's clock is ahead
> > +        // of ours by a small amount.
> > +        var timeCursor = NSDate(timeIntervalSinceNow: NSTimeInterval(minutes: 60))
> > +        let eventTimestampDeltaAllowance = NSTimeInterval(minutes: 60)
> Here's what my testing (on a 522) shows:
> 1) Square wave boluses don't get added into the history until they are done being delivered
> 2) When they are added, they have their start time as the timestamp
> 3) Other events can be added to history (changing alarm settings, etc) while the square wave is going
> 4) The ordering of events in the history is the order in which they are added, so temp basals end up with a timestamp that is out of order, and can be up to 8 hours out of order.
> 
> So I think I need to adjust the logic. This also has some implications for Looping and reservoir vs history processing. From a history view, you don't know if a square wave is being delivered.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>